### ISSUE TITLE
Inconsistent display of success message upon submission

### ISSUE THREAD

**Issue by KMJ1618** [MaayanLab/covid19_crowd_library#88]
Most of the time, when the user clicks the Submit button, a success message appears in a green box. However, sometimes no success message appears, and it's unclear whether such submissions have been successfully processed or not. If clicking the Submit button can consistently display this success message, that would be helpful, as I am hesitant to continue submitting content with no success messages...

**Comment by maxim-k** [comment]
It was fixed in yesterday's deployment. Now, the form is validated before submission and missing fields are highlighted.
Duplicate of #11.

**Comment by KMJ1618** [comment]
@maxim-k Now, I'm consistently not seeing any success message at all upon clicking the Submit button... Can this be fixed? I find it extremely helpful to see a success message to confirm that my submissions have been received...

**Comment by u8sand** [comment]
@KMJ1618 Investigating this, I was able to reproduce a similar issue and it looked to me like the request had an error submitting, so you're aware so when you didn't receive this message, your lists may have not gone through. We're sorry about that.

I've just fixed at least what I was able to reproduce, and @maxim-k will work on ensuring that in the case of errors they will be shown in place of the success message (they are already reported by the backend).

**Comment by KMJ1618** [comment]
@u8sand Thanks for the update; fortunately, once I stopped seeing success messages, I opted to stop submitting new gene sets, so I didn't end up wasting any effort. I'll look forward to interacting with the improved platform when I submit new gene sets this weekend.

**Comment by KMJ1618** [comment]
Update: I just tested this platform for the first time since the update was shared above, but I'm still not seeing a success message when I click the Submit button... I will refrain from submitting content until this issue is resolved.

**Comment by AviMaayan** [comment]
This is weird. I have no problems with submitting gene and drug sets... Try using Chrome.

**Comment by KMJ1618** [comment]
I tried both Chrome and Firefox...

**Comment by AviMaayan** [comment]
Can you try it while you are connected to the VPN?

**Comment by KMJ1618** [comment]
Ok, I could try that. I'll need to switch computers.

**Comment by KMJ1618** [comment]
I also cleared both browsers' caches, in case that was the issue, but the behavior hasn't changed.

One observation: In the submission form, before the Submit button is clicked, all of the entry boxes on the submission form have light grey boxes; in contrast, after the Submit button is clicked, all of the boxes' outlines change to green, and the text "Share contact information" changes from black to green. So, *something* is happening upon my clicking the Submit button.

**Comment by KMJ1618** [comment]
The above behavior occurred on a Windows 10 machine.

I also tested the site on a Mac, using both Chrome and Firefox browsers, and both logged into VPN and not logged in; in all 4 cases, the behavior on the Mac remains the same as described above.

**Comment by maxim-k** [comment]
> One observation: In the submission form, before the Submit button is clicked, all of the entry boxes on the submission form have light grey outlines; in contrast, after the Submit button is clicked, all of the boxes' outlines change to green, and the text "Share contact information" changes from black to green. So, _something_ is happening upon my clicking the Submit button.

@KMJ1618 it is form validation. If a field has green outline, it passed. More important - did you see any fields with red outline & suggestion?

**Comment by KMJ1618** [comment]
@maxim-k Thanks for the clarification. When I have repeatedly tried to submit the same example gene set (the format of which should be legit, as it has the same format as other previous successful submissions), no, there are no red boxes or text, only green boxes.

**Comment by u8sand** [comment]
@KMJ1618 Can you check your web brower's "Inspect" tool and let us know if you see any errors?

**Comment by KMJ1618** [comment]
This is the screenshot of 3 reported error messages following clicking Submit, using the Chrome browser.

![COVID-19 Platform_ChromeBrowser_ScreenshotOfErrorsFromInspectTool_4-14-20](https://user-images.githubusercontent.com/7283504/79274657-e4e7f600-7e72-11ea-8c1e-b42d7ee33ac2.PNG)

**Comment by u8sand** [comment]
@KMJ1618 Awesome, can you look at the network packet that reported the 500 and see what the response was

@maxim-k Are we still not reporting server errors on the UI? When the server has an error it provides an error message

**Comment by u8sand** [comment]
Because the server currently loses error messages (only sending them back the the user) the commit I made above will fix that, reporting errors in our logs as well.

**Comment by KMJ1618** [comment]
@u8sand >> can you look at the network packet that reported the 500 and see what the response was
I copied and pasted the text from the hyperlink “/covid19/genesets:1/” next to the 500 error message, which is the only interactive aspect of that error message that I see; this extensive text is attached. From an initial glace at this file, it appears that it includes extensive content from this database (possibly the full database?), with the names of various submitters visible, far beyond the 38-gene set that I had submitted for this single submission.

[Content for COVID-19 500 Error Message_v1_4-14-20.pdf](https://github.com/MaayanLab/covid19_crowd_library/files/4478105/Content.for.COVID-19.500.Error.Message_v1_4-14-20.pdf)

**Comment by u8sand** [comment]
That's not it, it's in the network tab. But don't worry, 
 we should have it in our error logs now, so I'll check it out.

**Comment by KMJ1618** [comment]
In case it's still relevant, attached is a screenshot of my Network tab from the browser Inspect tool, following the submission of a gene set using the Chrome browser.

<img width="1161" alt="NetworkTabFromInspectTool_4-14-20" src="https://user-images.githubusercontent.com/7283504/79286608-27b7c700-7e8f-11ea-81fa-97151cd31ed0.png">

**Comment by AviMaayan** [comment]
I was able to reproduce the problem and solve it. If you remove the text:
:
COVID-19 research efforts build on earlier research on Severe Acute Respiratory Syndrome (SARS) and Middle East Respiratory Syndrome (MERS), which also are caused by coronaviruses.
https://www.niaid.nih.gov/diseases-conditions/covid-19

from the "How is this set related to COVID-19 research?" text box it works. So it is something with the "How is this set related to COVID-19 research?" length or weird characters...

**Comment by KMJ1618** [comment]
A specific example of the issue that Avi described above:

I can't successfully submit this long "How is it related" description (which still fits within the entry box): 
The spike protein of SARS-CoV-2 is primed by TMPRSS2; SARS-CoV-2 uses the SARS-CoV receptor ACE2 for entry and the serine protease TMPRSS2 for S protein priming [Hoffmann et al. (2020) SARS-CoV-2 Cell Entry Depends on ACE2 and TMPRSS2 and Is Blocked by a Clinically Proven Protease Inhibitor, Cell, 181(2): pp.271-280].

...but if I reduce it to the following, the submission is successful:
The spike protein of SARS-CoV-2 is primed by TMPRSS2; SARS-CoV-2 uses the SARS-CoV receptor ACE2 for entry and the serine protease TMPRSS2 for S protein priming (Hoffmann et al. (2020) Cell, 181(2): pp.271-280).

**Comment by u8sand** [comment]
It seems pretty clear to me by this point that the Field is probably just
varchar(255) or something similar.. but frankly the server error message
should say this on the front end.

**Comment by maxim-k** [comment]
@KMJ1618 oh that's a relief. I finally can replicate and I know what's the problem - string length for full description is 255 characters.

**Comment by maxim-k** [comment]
@u8sand Jinx!
